auth/api/iam/types.go

```diff
@@ -0,0 +1,152 @@
+	// responseModeDirectPost signals the Authorization Server to POST the requested presentation definition to the provided response_uri
+	// https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-response-mode-direct_post
+	responseModeDirectPost responseMode = "direct_post"
+)
+
+var responseModesSupported = []responseMode{responseModeQuery, responseModeDirectPost}
+
+// grantType
+type grantType = string
+
+const (
+	// grantTypeAuthorizationCode is the default OAuth2 grant type
+	grantTypeAuthorizationCode grantType = "authorization_code"
+	// grantTypeVPToken is used to present a vp_token in exchange for an access_token in service-to-service flows
+	// TODO: EBSI inspired flow that is not standardized
+	// 		 https://api-conformance.ebsi.eu/docs/ct/verifiable-presentation-exchange-guidelines-v3#service-to-service-token-flow
+	grantTypeVPToken grantType = "vp_token"
+	// grantTypePreAuthorizedCode is defined in the pre-authorized_code flow of OpenID4VCI
+	// https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-sub-namespace-registration
+	grantTypePreAuthorizedCode grantType = "urn:ietf:params:oauth:grant-type:pre-authorized_code"
+)
+
+var grantTypesSupported = []grantType{grantTypeAuthorizationCode, grantTypeVPToken, grantTypePreAuthorizedCode}
+
+// algValuesSupported contains a list of supported cipher suites for jwt_vc_json & jwt_vp_json presentation formats
+// Recommended list of options https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms
+// TODO: validate list, should reflect current recommendations from https://www.ncsc.nl
+var algValuesSupported = []string{"PS256", "PS384", "PS512", "ES256", "ES384", "ES512"}
+
+// proofTypesSupported contains a list of supported cipher suites for ldp_vc & ldp_vp presentation formats
+// Recommended list of options https://w3c-ccg.github.io/ld-cryptosuite-registry/
+var proofTypesSupported = []string{"JsonWebSignature2020"}
+
+// vpFormatsSupported defines the supported formats and is used in the
+//   - Authorization Server's metadata field `vp_formats_supported`
+//   - Client's metadata field `vp_formats`
+//
+// TODO: spec is very unclear about this part.
+//
+//	It refers to the OpenID4VCI spec for definition of valid credential formats but then also defines its own,
+//	but neither is followed in the examples.
+//
+// OpenID4VCI defined formats are jwt_vc_json, jwt_vc_json-ld, ldp_vc
+//
+// OpenID4VP defines:
+//   - jwt_vc_json & jwt_vp_json   that should use values defined on https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms
+//   - ldp_vc      & ldp_vp		that should use valued defined on https://w3c-ccg.github.io/ld-cryptosuite-registry/
+//
+// but the examples then seem to be a mix and match of everything.
+var vpFormatsSupported = map[string]map[string][]string{
+	"jwt_vp": {"algValuesSupported": algValuesSupported},
```

Comment:
do we really support this one?, also, the comment above says jwt_vc_json?

Reply:
changed to `jwt_vp_json`. I based this on #2357, but could be incorrect?